client/ui/models/apiServicesModel.cpp

```diff
@@ -209,10 +217,46 @@ QHash<int, QByteArray> ApiServicesModel::roleNames() const
     roles[ServiceDescriptionRole] = "serviceDescription";
     roles[IsServiceAvailableRole] = "isServiceAvailable";
     roles[SpeedRole] = "speed";
-    roles[WorkPeriodRole] = "workPeriod";
+    roles[TimeLimitRole] = "timeLimit";
     roles[RegionRole] = "region";
     roles[FeaturesRole] = "features";
     roles[PriceRole] = "price";
+    roles[EndDateRole] = "endDate";
 
     return roles;
 }
+
+ApiServicesModel::ApiServicesData ApiServicesModel::getApiServicesData(const QJsonObject &data)
```

Comment:
It would be great to add input JSON validation when accessing values in this method.

Reply:
added check here https://github.com/amnezia-vpn/amnezia-client/pull/1261/files#diff-f656f20f69d23c37a36b3db56c5206866151ad3dca11db74f8ecc342a122d778R381